litestar/connection/websocket.py

```diff
@@ -167,8 +167,6 @@ async def receive_data(self, mode: WebSocketMode) -> str | bytes:
         event = cast("WebSocketReceiveEvent | WebSocketDisconnectEvent", await self.receive())
         if event["type"] == "websocket.disconnect":
             raise WebSocketDisconnect(detail="disconnect event", code=event["code"])
-        if self.connection_state == "disconnect":
-            raise WebSocketDisconnect(detail=DISCONNECT_MESSAGE)  # pragma: no cover
```

Comment:
FWICT - I don't think this code is reachable - which probably explains the pragma.

This method calls `self.receive()` which is really `self.receive_wrapper()`, which will either set `self.connection_state` to `"disconnect"` and return the message if its type is `websocket.disconnect`, or (prior to this PR) raise `WebsocketException` if `self.connection_state == "disconnect"` is already true.